File: cli/internal/agentauthz/env.go
package agentauthz

import (
	"strings"
)

var AuthSourceEnvKeys = []string{
	EnvAuthBlob,
	EnvAuthBlobFile,
	EnvAuthUserID,
	EnvRequesterContextDir,
}

func AuthSourceEnvPresent(env map[string]string) bool {
	if env == nil {
		env = environMap()
	}
	for _, key := range AuthSourceEnvKeys {
		if strings.TrimSpace(env[key]) != "" {
			return true
		}
	}
	return false
}

func ScrubAuthSourceEnvCommand(command string) string {
	return "unset " + strings.Join(AuthSourceEnvKeys, " ") + "; " + command
}

func ScrubAuthSourceEnvPowerShellCommand(command string) string {
	paths := make([]string, 0, len(AuthSourceEnvKeys))
	for _, key := range AuthSourceEnvKeys {
		paths = append(paths, "Env:"+key)
	}
	return "Remove-Item " + strings.Join(paths, ",") + " -ErrorAction SilentlyContinue; " + command
}

File: packages/data-harness-cli/src/lib/authz/env.js
import { AUTH_SOURCE_ENV_KEYS } from "./constants.js";

export function environMap(env = process.env) {
  /** @type {Record<string, string>} */
  const out = {};
  for (const [key, value] of Object.entries(env)) {
    if (value != null) out[key] = String(value);
  }
  return out;
}

export function authSourceEnvPresent(env) {
  const map = env || environMap();
  return AUTH_SOURCE_ENV_KEYS.some((key) => String(map[key] || "").trim() !== "");
}

export function scrubAuthSourceEnvCommand(command) {
  return `unset ${AUTH_SOURCE_ENV_KEYS.join(" ")}; ${command}`;
}

export function scrubAuthSourceEnvPowerShellCommand(command) {
  const paths = AUTH_SOURCE_ENV_KEYS.map((key) => `Env:${key}`);
  return `Remove-Item ${paths.join(",")} -ErrorAction SilentlyContinue; ${command}`;
}
